app/migrations/Version20160426000000.php

```diff
@@ -0,0 +1,156 @@
+<?php
+/**
+ * @package     Mautic
+ * @copyright   2016 Mautic Contributors. All rights reserved.
+ * @author      Mautic
+ * @link        http://mautic.org
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\Migrations;
+
+use Doctrine\DBAL\Migrations\SkipMigrationException;
+use Doctrine\DBAL\Schema\Schema;
+use Mautic\CoreBundle\Doctrine\AbstractMauticMigration;
+use Mautic\LeadBundle\Entity\DoNotContact;
+
+/**
+ * Universal DNC Migration
+ */
+class Version20160426000000 extends AbstractMauticMigration
+{
+    private $leadIdIdx;
+    private $leadIdFk;
+
+    /**
+     * @param Schema $schema
+     *
+     * @throws SkipMigrationException
+     * @throws \Doctrine\DBAL\Schema\SchemaException
+     */
+    public function preUp(Schema $schema)
+    {
+        if ($schema->hasTable($this->prefix.'lead_donotcontact')) {
+            throw new SkipMigrationException('Schema includes this migration');
+        }
+
+        $this->leadIdIdx = $this->generatePropertyName($this->prefix . 'lead_donotcontact', 'idx', array('lead_id'));
+        $this->leadIdFk  = $this->generatePropertyName($this->prefix . 'lead_donotcontact', 'fk', array('lead_id'));
+    }
+
+    /**
+     * @param Schema $schema
+     */
+    public function mysqlUp(Schema $schema)
+    {
+
+        $sql = <<<SQL
+CREATE TABLE `{$this->prefix}lead_donotcontact` (
+  `id` int(11) NOT NULL AUTO_INCREMENT,
+  `lead_id` int(11) DEFAULT NULL,
+  `channel` varchar(255) COLLATE utf8_unicode_ci NOT NULL,
+  `channel_id` int(11) DEFAULT NULL,
+  `date_added` datetime NOT NULL COMMENT '(DC2Type:datetime)',
+  `reason` smallint NOT NULL,
+  `comments` longtext COLLATE utf8_unicode_ci,
+  PRIMARY KEY (`id`),
+  KEY `{$this->leadIdIdx}` (`lead_id`),
+  KEY `{$this->prefix}dnc_reason_search` (`reason`),
+  CONSTRAINT `{$this->leadIdFk}` FOREIGN KEY (`lead_id`) REFERENCES `{$this->prefix}leads` (`id`) ON DELETE CASCADE
+) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;
+SQL;
+
+        $this->addSql($sql);
+    }
+
+    /**
+     * @param Schema $schema
+     */
+    public function postgresqlUp(Schema $schema)
+    {
+        $this->addSql("CREATE SEQUENCE {$this->prefix}lead_donotcontact_id_seq INCREMENT BY 1 MINVALUE 1 START 1");
+
+        $sql = <<<SQL
+CREATE TABLE {$this->prefix}lead_donotcontact (
+  id INT NOT NULL, 
```

Comment:
This line should be `id SERIAL NOT NULL,` and then the postgreSQL migration starts to work.